fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -191,6 +193,7 @@ void logicalRuleRewrite(Memo memo, TaskContext rootTaskContext) {
 
         ruleRewriteIterative(memo, rootTaskContext, RuleSetType.MULTI_DISTINCT_REWRITE);
         ruleRewriteIterative(memo, rootTaskContext, RuleSetType.SUBQUERY_REWRITE);
+        ruleRewriteIterative(memo, rootTaskContext, new PushDownProjectionToCTEAnchorRule());
```

Comment:
move the rule to RuleSet.PRUNE_PROJECT is enough？

Reply:
Exactly

Reply:
move it after modify the pattern